src/composed/shared.rs

```diff
@@ -78,12 +79,21 @@ pub trait Deserializable: Sized {
     /// Parse a list of compositions in raw byte format.
     fn from_bytes_many<'a>(bytes: impl Read + 'a) -> Box<dyn Iterator<Item = Result<Self>> + 'a> {
         let packets = PacketParser::new(bytes).filter_map(|p| {
-            // for now we are skipping any packets that we failed to parse
-            if p.is_ok() {
-                p.ok()
-            } else {
-                warn!("skipping packet: {:?}", p);
-                None
+            match &p {
+                Ok(_) => p.ok(),
+                Err(e) => {
+                    if let InvalidPacketContent(b) = e {
+                        let err: &Error = b; // unbox
+                        if let Error::Unsupported(_e) = err {
+                            // Error::Unsupported signals packet versions etc that we explicitly detected and can safely ignore
+                            warn!("skipping unsupported packet: {:?}", p);
+                            return None;
+                        }
+                    }
+
+                    // Fail on packets that we failed to parse for unspecific reasons
+                    panic!("unexpected packet data: {p:?}");
```

Comment:
we really shouldn't panic by design, especially not on content controlled by an external party

Reply:
Fully agreed, of course.

I've commited an update to this draft with explicit error handling instead of those (placeholder) panics.
My current version of an attempt at orderly error handling logic is in 54e4c162c30987d398a6f16caf36c447a89b0063